cpp/ql/src/Likely Bugs/Arithmetic/IntMultToLong.cpp

```diff
@@ -4,3 +4,5 @@ long j = i * i; //Wrong: due to overflow on the multiplication between ints,
 
 long k = (long) i * i; //Correct: the multiplication is done on longs instead of ints, 
                        //and will not overflow
+
+long l = static_cast<long>(i * i); //Correct: modern C++
```

Comment:
Actually should this be `static_cast<long>(i) * i` or `static_cast<long>(i) * static_cast<long>(i)`?  As it is the `int`s are multiplied, overflow, and are then cast.

Reply:
That I'm not able to answer, but it did made CodeQL happy...

Reply:
I'm reading the comment `//Correct: modern C++` as saying "a test like the one above this one", so (since line 5 is equivalent to `long k = ((long) i) * i`) I would guess that line 8 is meant to be `static_cast<long>(i) * i`.

Reply:
It does, I think that's because the query assumes you 'know what you're doing' if there's a cast anywhere in the expression.  But it should be cast before multiplying.

Reply:
```suggestion
long l = static_cast<long>(i) * i; //Correct: modern C++
```